WinWcpRawIO: Add rec_datetime to header dict and block annotation from file

danielparthier opened the pull request:
In the current version of WinWcpRawIO only limited information was read from the file. One missing part was the recording time (rec_datetime) which was now added to the header dict and block annotation.

zm711 reviewed (commented):
Thanks for the PR @danielparthier ! Seems like our test file has a different spec where this doesn't exist so could we ask two things:

1) could you fix your PR to account for whatever version we have with our test repo to account for the fact not all have "RTIME". We often find with hardware devs they change the keys between versions. 
2) could you provide a < 10MB test file of the version you're using so we can include it in our testing suite?

danielparthier commented:
I updated the code and added a pull request to the gin repository. Gin didn't like to be run in terminal on my side so I am not 100% it worked because I had to go via the web interface.

zm711 commented:
The other option is you can send the file here via zip and we can download it and then upload ourselves. The gin branching mechanism is hard to deal with so I'm not sure how you got the pegasus v2.2 since they aren't on the main branch, but my guess is the client accidentally synced you to the wrong branch.

danielparthier commented:
Yes the zip file sounds like a good option/short-cut. Still not sure how it managed to do that. Thanks for the help!
[File_winwcp_2.zip](https://github.com/user-attachments/files/20839068/File_winwcp_2.zip)

h-mayorquin commented:
Hi, @danielparthier

I can add the file for you. Can you write some short info about the file (channels, size, purpose, etc) and about yourself (if you want) so we add it to the Readme for provenance.

h-mayorquin commented:
Some example:
https://gin.g-node.org/NeuralEnsemble/ephy_testing_data/src/master/plexon/README.md

h-mayorquin commented:
Thanks. Here you go guys:

https://gin.g-node.org/NeuralEnsemble/ephy_testing_data/pulls/159

@zm711

h-mayorquin commented:
@danielparthier the test files have been merged. You can add a tests now here and the CI should be able to fetch it.

zm711 commented:
Let us know if you have trouble with the testing. It should happen at both the rawio and io level. Thanks @h-mayorquin!

danielparthier commented:
> @danielparthier the test files have been merged. You can add a tests now here and the CI should be able to fetch it.

The new test file is now included in the iotest and rawiotest (https://github.com/NeuralEnsemble/python-neo/pull/1733/commits/52c46a5250c10f3ca2852989073c164b5f9d27ce)

Thanks guys for the quick feedback!

zm711 commented on neo/rawio/winwcprawio.py:
```diff
@@ -164,10 +172,12 @@ def _parse_header(self):
         self.header["signal_channels"] = signal_channels
         self.header["spike_channels"] = spike_channels
         self.header["event_channels"] = event_channels
+        self.header["rec_datetime"] = rec_datetime
```
Thanks! :)

danielparthier commented:
> Thanks @danielparthier!
> 
> Do you want to be included in our author list. If so could you give the name and affiliation you want (I assume it'll be the same as for the file you shared, but I always prefer to have an official confirmation :) )

Yes that would be the same as in the file. Thanks again for the quick help!
